client/state/reducer-registry/index.js

```diff
@@ -0,0 +1,152 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { reduce, uniq } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { DESERIALIZE, DESERIALIZE_PART, SERIALIZE } from 'state/action-types';
+import { createReducer, withSchemaValidation } from 'state/utils';
+
+/**
+ * Module variables
+ */
+const registeredReducers = Object.create( null );
+let defaultStoreDispatch;
+
+/**
+ * Pleaseholder reducer keeps undeserialized data in the in-memory state tree
+ * so that we could keep the serialized data persistent before loading the actual
+ * reducers and its schema, and allow the state to be deserialized when they are
+ * loaded (see below.)
+ */
+const placeholderReducer = createReducer( null, {
+	[ SERIALIZE ]: state => {
+		return state._initialState;
+	},
+	[ DESERIALIZE ]: state => {
+		return {
+			_initialState: state,
+		};
+	},
+} );
+
+/**
+ * This method returns a reducer that respond to DESERIALIZE_PART action type,
+ * so that we could deserialize the state with the newly loaded reducer.
+ * @param  {Function} reducer The newly loaded reducer
+ * @return {Function}         Reducer that responds to DESERIALIZE_PART action type.
+ */
+function getDeserializeReducer( reducer ) {
+	return createReducer( null, {
+		[ SERIALIZE ]: state => {
+			return state._initialState;
+		},
+		[ DESERIALIZE_PART ]: state => {
+			const initialState = state == null ? undefined : state._initialState;
```

Comment:
Lodash get is a bit cleaner here.
`const initialState = get( state, '_initialState', undefined );`